sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/LongOffset.scala

```diff
@@ -22,8 +22,18 @@ package org.apache.spark.sql.execution.streaming
  */
 case class LongOffset(offset: Long) extends Offset {
 
+  override val json = offset.toString
+
   def +(increment: Long): LongOffset = new LongOffset(offset + increment)
   def -(decrement: Long): LongOffset = new LongOffset(offset - decrement)
 
   override def toString: String = s"#$offset"
 }
+
+object LongOffset {
+
+  def apply(serialized: Offset) : LongOffset = new LongOffset(serialized.json.toLong)
+
+}
```

Comment:
Not a strict guideline i guess, its matter of the taste of the reviewer quite a bit. For me i guess where there is no need to visually introduce a gap, there is no need for an empty line. For example: between `def apply` and `}` there no need to introduce a separation.